docs/benchmark/ftlib_benchmark.md

```diff
@@ -0,0 +1,171 @@
+managed by the same tailored Kubernetes system. In those experiments,
+we adopt ResNet50 and MobileNetV2 models to test the performance of
+computation and communication. ResNet50 is a computation-intensive
+model and MobileNetV2 is a communication-intensive model.
+
+## Minikube Cluster
+
+We start a local Kubernetes cluster on macOS by the command:
+
+```bash
+minikube start --vm-driver=hyperkit --cpus 2 --memory 6144 --disk-size=50gb
+```
+
+The experiment parameters are:
+
+| Experiment parameter | Value    |
+| -- | --- |
+| Batch size | 64 |
+| Batches per task | 50 |
+| Images per task | 3220 |
+| Dataset | CIFAR-10 |
+| Input shape | (32, 32, 3) |
+| Output classes| 10 |
+| Worker resource| cpu=0.3,memory=2048Mi,ephemeral-storage=1024Mi|
+
+### ResNet50 with CIFAR-10 on a Minikube Cluster
+
+There are 214 parameter tensors in ResNet50 for CIFAR-10 dataset. And
+those tensor totally have 23,555,082 trainable parameters.
+
+| Workers |  computation/communication  |  Speed   |  Speedup Ratio |
+| -- | --------------------------- | -------- | ------ |
+| 1  |  -  | 3.1 images/s  |  1  |
+| 2  | 10: 1 | 5.65 images/s | 1.82 |
+
+### MobileNetV2 with CIFAR-10 on a Minikube Cluster
+
+There are 158 parameter tensors in ResNet50 for CIFAR-10 dataset. And
+those tensors have 2,236,682 trainable parameters.
+
+| Workers   |  computation/communication  |  Speed   |  Speedup Ratio |
+| -- | --------------------------- | -------- | ------ |
+| 1  |   -  | 29 images/s  |  1  |
+| 2  | 10: 3 | 44.7 images/s | 1.54 |
+| 3  | 10: 6 | 57.2 images/s  | 1.97 |
+
+## On-premise CPU Cluster Managed by a Tailored Kubernetes System
+
+There are more resource on an on-premise CPU cluster, so we can turn up the
+worker resource to do experiments. And, we also use CIFAR-10 dataset and
+assign more CPUs and memory for the worker. The experiment parameters are:
+
+| Experiment parameter | Value |
+| -- | --- |
+| Batch size | 64 |
+| Batches per task | 50 |
+| Images per task | 3220 |
+| Dataset | CIFAR-10 |
+| Input shape | (32, 32, 3) |
+| Output classes| 10 |
+| Worker resource| cpu=4,memory=8192Mi,ephemeral-storage=1024Mi|
+
+### ResNet50 with CIFAR-10 on an On-premise CPU Cluster
+
+| Workers   | communication   |  Speed   |  Speedup Ratio |
+| -- | --------------------------- | -------- | ------ |
+| 1  |   0%  | 26.7 images/s  |  1  |
+| 2  | 18% | 41 images/s | 1.57 |
+| 4  | 25% | 68.4 images/s  | 2.56 |
+| 8  | 32% | 123 images/s | 4.61 |
+
+### MobileNetV2 with CIFAR-10 on an On-premise CPU Cluster
+
+| Workers   | communication   |  Speed   |  Speedup Ratio |
+| -- | --------------------------- | -------- | ------ |
+| 1  |   0%  | 353.6 images/s  |  1  |
+| 2  | 24% | 503 images/s | 1.42 |
+| 4  | 44.7% | 680 images/s  | 1.92 |
+| 8  | 66.7% | 648 images/s  | 1.83 |
+
+For the above experiments of CPU, we find that the speed-up ratio is better if
+the model is more complex. Because the computation on CPUs is
```

Comment:
Please be specific with "complex" -- do you want to say that the model has more layers, more parameters, or something else?

Reply:
Yes, "complex" stands for more layers and more parameters.